std/functional.d

```diff
@@ -260,6 +210,124 @@ unittest
     static assert(!is(typeof(binaryFun!FuncObj)));
 }
 
+/**
+Transforms a string representing an expression into a n-ary function.
+$(UL
+    $(LI The number of parameters is given by `argc`. If no explicit
+        `argc` is given, it is inferred to be `paramNames.length`.)
+    $(LI Explicit names for the function parameters may be given as
+        `paramNames`. If more than `argc` names are supplied, the
+        extras will be ignored.)
+    $(LI The letters (`a` through `z`) are used as default names
+        for up to the first 26 function parameters.)
+    $(LI `args[index]` can always be used to access any parameter by index,
+        regardless of whether it has been given a name.)
+)
+*/
+template stringLambda(alias fun, size_t argc, paramNames...)
+    if (paramNames.length == 0 || allSatisfy!(isSomeString, typeof(paramNames)))
+{
+    static if (is(typeof(fun) : string))
+    {
+        static if (!fun._ctfeMatchNary(paramNames))
+        {
+            import std.traits, std.typecons, std.typetuple;
+            import std.algorithm, std.conv, std.exception, std.math, std.range, std.string;
+        }
+
+        auto stringLambda(ArgTypes...)(auto ref ArgTypes args)
+            if (ArgTypes.length == argc)
+        {
+            mixin(stringLambdaAliases!(argc, paramNames));
+            return mixin(fun);
+        }
+    }
+    else static if (needOpCallAlias!fun)
+    {
+        // Issue 9906
+        alias stringLambda = fun.opCall;
+    }
+    else
+    {
+        alias stringLambda = fun;
+    }
+}
+
+/// ditto
+template stringLambda(alias fun, paramNames...)
+    if (allSatisfy!(isSomeString, typeof(paramNames)))
+{
+    alias stringLambda = stringLambda!(fun, paramNames.length, paramNames);
+}
+
+// Generate stringLambda parameter aliases mixin string.
```

Comment:
A better comment, to prevent some well-meaning person from de-optimizing this later:
```
/* The stringLambda parameter aliases mixin depends only on argc and paramNames, not fun. Therefore,
   compilation can be sped up by generating it at global scope and memoizing it as a template,
   instead of regenerated it with every stringLambda instantiation.*/
```